src/Microsoft.AspNet.Mvc.Abstractions/ModelBinding/Validation/ClientModelValidationContext.cs

```diff
@@ -5,7 +5,7 @@
 
 namespace Microsoft.AspNet.Mvc.ModelBinding.Validation
 {
-    public class ClientModelValidationContext
+    public class ClientModelValidationContext : ModelValidationContextBase
```

Comment:
Refactoring a few properties into `ModelValidationContextBase` creates a weird mishmash of semantics. If the base class is absolutely required, consistently use the get-only approach. If `ModelValidationContext` absolutely requires get/set properties, remove the base class and pass `ModelMetadata` and `IModelMetadataProvider` into `GetErrorMessage()`.

In general I'm not a big fan of adding `ModelValidationContextBase`.